includes/class-wc-payments-webhook-processing-service.php

```diff
@@ -331,6 +341,20 @@ private function process_webhook_payment_intent_succeeded( $event_body ) {
 		$charge_id     = $this->read_webhook_property( $charges_data[0], 'id' );
 
 		$this->order_service->mark_payment_completed( $order, $intent_id, $intent_status, $charge_id );
+
+		// Send the customer a card reader receipt if it's an in person payment type.
+		if ( array_key_exists( 'payment_method_details', $charges_data[0] ) && 'card_present' === $this->read_webhook_property( $charges_data[0]['payment_method_details'], 'type' ) ) {
+			$wcpay_gateway     = WC_Payments::get_gateway();
```

Comment:
Also, please inject the gateway via constructor (or let me know if there are issues with this).

Reply:
Done